firedrake/slate/slac/kernel_builder.py

```diff
@@ -219,6 +220,6 @@ def generate_expr_data(expr):
             # acting coefficient. For inverses, we may declare additional
             # temporaries (depending on reference count).
             if isinstance(tensor, (Action, Inverse)):
-                aux_exprs.setdefault(tensor)
+                aux_exprs.append(tensor)
```

Comment:
If you aren't sure that there will be no duplications, you may want to add an `aux_exprs = list(OrderedDict.fromkeys(aux_exprs))` line after the for loop.